x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/enrich/50_data_stream.yml

```diff
@@ -0,0 +1,111 @@
+---
+setup:
+  - skip:
+      features: allowed_warnings
+  - do:
+      allowed_warnings:
+        - "index template [enrich-template] has index patterns [my-ds] matching patterns from existing older templates [global] with patterns (global => [*]); this template [enrich-template] will take precedence during new index creation"
+      indices.put_index_template:
+        name: enrich-template
+        body:
+          index_patterns: [my-ds]
+          template:
+            settings:
+              index.number_of_replicas: 0
+          data_stream: {}
+
+  - do:
+      bulk:
+        refresh: true
+        index: my-ds
+        body:
+          - '{"create": {}}'
+          - '{"@timestamp": "2022-01-01", "baz": "quick", "a": "brown", "b": "fox"}'
+          - '{"create": {}}'
+          - '{"@timestamp": "2022-01-01", "baz": "lazy", "a": "dog"}'
+
+  - do:
+      enrich.put_policy:
+        name: test_policy
+        body:
+          match:
+            indices: ["my-ds"]
+            match_field: baz
+            enrich_fields: ["a", "b"]
+
+  - do:
+      enrich.execute_policy:
+        name: test_policy
+
+  - do:
+      ingest.put_pipeline:
+        id: test_pipeline
+        body:
+          processors:
+            - enrich:
+                policy_name: test_policy
+                field: baz
+                target_field: target
+
+---
+teardown:
+  - do:
+      ingest.delete_pipeline:
+        id: test_pipeline
+
+  - do:
+      enrich.delete_policy:
+        name: test_policy
+
+---
+"enrich documents over _bulk via a data stream":
```

Comment:
Would we also want a test that ensures that the enrich policy is able to fetch data from non-write index backing indices? Or is that overkill?

I also don't see any YAML REST tests (or integration tests in general for that matter) that cover using aliases in enrich. Is that something worth adding if we have a specific test for data streams as well?

Reply:
That's a good idea, I've added a yaml test for an enrich policy that uses an alias.